app/backend/src/couchers/models.py

```diff
@@ -653,3 +653,262 @@ class InitiatedUpload(Base):
+    """
+
+    __tablename__ = "clusters"
+
+    id = Column(BigInteger, primary_key=True)
+    name = Column(String, nullable=False)
+    main_page_id = Column(BigInteger, ForeignKey("pages.id"), nullable=False, unique=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+
+class NodeClusterAssociation(Base):
+    """
+    NodeClusterAssociation, grouping of nodes
+    """
+
+    __tablename__ = "node_cluster_associations"
+    __table_args__ = (UniqueConstraint("node_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    node_id = Column(BigInteger, ForeignKey("nodes.id"), nullable=False)
+    cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False)
+
+
+class GroupRole(enum.Enum):
+    member = 1
+    admin = 2
+
+
+class ClusterSubscription(Base):
+    """
+    ClusterSubscription of a user
+    """
+
+    __tablename__ = "cluster_subscriptions"
+    __table_args__ = (UniqueConstraint("user_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    user_id = Column(BigInteger, ForeignKey("users.id"), nullable=False)
+    cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False)
+    role = Column(Enum(GroupRole), nullable=False)
+    joined = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+    left = Column(DateTime(timezone=True))
+
+
+class ClusterPageAssociation(Base):
+    """
+    pages related to clusters
+    """
+
+    __tablename__ = "cluster_page_associations"
+    __table_args__ = (UniqueConstraint("page_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    page_id = Column(BigInteger, ForeignKey("pages.id"), nullable=False)
+    cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False)
+
+
+class PageType(enum.Enum):
+    main_page = 1
+    point_of_interest = 2
+    guide = 3
+    event = 4
+
+
+class Page(Base):
+    """
+    similar to a wiki page
+    """
+
+    __tablename__ = "pages"
+
+    id = Column(BigInteger, primary_key=True)
+
+    type = Column(Enum(PageType), nullable=False)
+    thread_id = Column(BigInteger, ForeignKey("threads.id"), nullable=False, unique=True)
+    creator_user_id = Column(BigInteger, ForeignKey("users.id"), nullable=False)
+    owner_user_id = Column(BigInteger, ForeignKey("users.id"), nullable=False)
+    owner_cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False, unique=True)
+
+
+class PageVersion(Base):
+    """
+    version of page content
+    """
+
+    __tablename__ = "pageversions"
+
+    id = Column(BigInteger, primary_key=True)
+
+    page_id = Column(BigInteger, ForeignKey("pages.id"), nullable=False)
+    editor_user_id = Column(BigInteger, ForeignKey("users.id"), nullable=False)
+    title = Column(String, nullable=False)
+    content = Column(String, nullable=False)
+    geotag = Column(Geometry(geometry_type="POINT", srid=4326), nullable=False)
```

Comment:
Let's stay consistent and call this `geom`.

Also, I'm thinking, maybe we'd want general geometries here? E.g. multipoint (guide about multiple restaurants?)? Polygon? I'm not sure.

Reply:
fixed